src/com/maddyhome/idea/vim/group/SearchGroup.java

```diff
@@ -647,6 +654,43 @@ else if (start >= offset) {
     return max;
   }
 
+  @Nullable
+  public static TextRange findCurrentOrNextSearch(Editor editor, int count, boolean forwards) {
+    String lastSearch = VimPlugin.getSearch().getLastSearch();
+    if (lastSearch == null) {
+      return null;
+    }
+    int currentPos = editor.getCaretModel().getOffset();
+    TextRange nextRange = new TextRange(currentPos, currentPos);
+    if (CommandState.getInstance(editor).getMode() == CommandState.Mode.VISUAL) {
+      return findNext(editor, lastSearch, nextRange, forwards, count);
+    }
+    else {
+      int startOffset = nextRange.getStartOffset();
+      nextRange = findCurrent(editor, lastSearch, startOffset)
+        .orElse(findNext(editor, lastSearch, startOffset, false, forwards));
+      if (nextRange == null) {
+        return null;
+      }
+      return findNext(editor, lastSearch, nextRange, forwards, count - 1);
+    }
+  }
+
+  @Nullable
+  private static TextRange findNext(Editor editor,
+                                    String lastSearch,
+                                    TextRange nextRange,
+                                    boolean forwards,
+                                    int count) {
+    for (int i = 0; i < count; i++) {
+      nextRange = findNext(editor, lastSearch, Math.max(nextRange.getStartOffset() - (forwards?0:1),0), false, forwards);
```

Comment:
I'm concerned about the performance in this place. It is not good to make n searches. Could you check `n` command implementation? It seems to me that you can use functions from there and there you can specify the count.